### DISCUSSION TITLE
Helmchart object returning "404 not found" when trying to pull chart versions

### DISCUSSION THREAD

**Discussion by soumyabk** [fluxcd/flux2#3382] [Q&A] [answered]
we currently have a helmchart object that fails to pull the helmchart resources. 
The helmrepository object seems to sync successfully and if I provide a chart version that does not exist, the helm chart object shows the error that the chart version does not exist. 

Helmrepository object:
```
kubectl -n flux-system get helmrepository artifactory
NAME          URL                                                          AGE   READY   STATUS
artifactory   https://<repo_url>/artifactory/<repo_name>  28h   True    stored artifact for revision '07aa3099ff005735c9a413440761161718eccf4721ce2a20e2a58ebacf67f19a'
```

Helmc hart object:
```
kubect -n flux-system get helmchart
NAME                      CHART         VERSION   SOURCE KIND      SOURCE NAME   AGE   READY   STATUS
flux-system-scanner   scanner   0.0.4     HelmRepository   artifactory   11m   False   chart pull error: failed to download chart for remote reference: failed to fetch https://<repo_url>:443/artifactory/api/helm/<repo_name>/scanner/scanner-0.0.4.tgz : 404 Not Found
```

However when provide a version that does not exist, it throws this error:
```
kubectl -n flux-system get helmchart
NAME                      CHART         VERSION   SOURCE KIND      SOURCE NAME   AGE   READY   STATUS
flux-system-scanner   scanner   0.0.6     HelmRepository   artifactory   13m   False   invalid chart reference: failed to get chart version for remote reference: no 'scanner' chart with version matching '0.0.6' found
```

I suspect that something is causing the request to pull the chart to not pass the right credentials or not be able to download the chart. Please let me know if you have any ideas on what might be causing this.

**Comment by kingdonb** [comment, marked as answer]
Can you provide more information about the Flux resources and the chart?

Namely: Is there any chart dependency? What is the chart name? (Does it have a slash in it?)

Is this a regression from a previous release of Flux, or is this a new chart you've never used with Flux before? (Also, what is the output of `flux version` and `flux check`, so we can know if you're on the latest versions?)

There are some cases where dependencies will not receive the credentials from the parent if there is a mismatching hostname for example, or if you have not enabled passCredentials: https://fluxcd.io/flux/components/source/helmrepositories/#pass-credentials

**Reply by soumyabk** [reply]
seems like using pass-credentials fixed the problem for us. Seems like our helm repository now switched to multi-az deployment so that is why it stopped working. Thank you so much for this!

**Comment by chaitu24** [comment]
Hi @soumyabk @kingdonb  , I'm using this approach. But still getting the error

My helmrepo

`
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: HelmRepository
metadata:
  name: grafana
  namespace: flux-system
spec:
  interval: 1ms
  passCredentials: true
  provider: generic
  secretRef:
    name: helm-gra
  timeout: 60s
  url: https://raw.githubusercontent.com/org/repo/gh-pages/

`

`
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: grafana-as
  namespace: flux-system
spec:
  chart:
    spec:
      chart: grafana     
      reconcileStrategy: ChartVersion
      sourceRef:
        kind: HelmRepository
        name: grafana
  interval: 10s
  releaseName: grafana-rs
  targetNamespace: grafana-prep

`
